include/zephyr/devicetree.h

```diff
@@ -2219,6 +2231,22 @@
  */
 #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
 
+/**
+ * @brief 64-bit version of a DT_REG_ADDR macro
+ *
+ * This macro adds ULL postfix to the address value returned
+ * by DT_REG_ADDR macro.
+ * Generally DT_REG_ADDR_U64 macro could be used interchangeably with
+ * DT_REG_ADDR macro with one exception: it couldn't be used in
+ * linker scripts.
+ * The general rule of thumb is to use 64-bit version when the
+ * appearance of 64-bit address is possible and when it's not
+ * conflicting with linker scripts
+ * @param node_id node identifier
+ * @return node's register block address
```

Comment:
```suggestion
/**
 * @brief 64-bit version of DT_REG_ADDR()
 *
 * This macro version adds the appropriate suffix for 64-bit unsigned integer literals.
 * Note that this macro is equivalent to DT_REG_ADDR() in linker context.
 *
 * @param node_id node identifier
 * @return node's register block address
```

Description needs to be more concise, a proposal above ^

Reply:
Thanks, will change it

Reply:
Hi @gmarull , I've addressed your comments